### PR TITLE
add rocksdb options manual_wal_flush support

### PR DESCRIPTION
**Pull request by javaforfun**

### CODE REVIEW

File: tests/test_rocksdb_options.rs [hunk 1]
```diff
@@ -412,6 +425,18 @@ fn test_allow_concurrent_memtable_write() {
...
 #[test]
+fn test_manual_wal_flush() {
+    let path = TempDir::new("_rust_rocksdb_manual_wal_flush").expect("");
+    let mut opts = DBOptions::new();
+    opts.create_if_missing(true);
+    opts.manual_wal_flush(true);
+    let db = DB::open(opts, path.path().to_str().unwrap()).unwrap();
+    for i in 0..200 {
+        db.put(format!("k_{}", i).as_bytes(), b"v").unwrap();
+    }
```
Comment (zhangjinpeng1987) [inline tests/test_rocksdb_options.rs]: Weather it will lose some data when reopen the DB?

Comment (siddontang) [reply tests/test_rocksdb_options.rs]: I think we can check the data again after re-opening the DB.

**Comment by siddontang** [comment]
Please fix the conflict.

**Comment by siddontang** [comment]
PTAL @huachaohuang @zhangjinpeng1987

File: tests/test_rocksdb_options.rs (hunk 1 above)
Comment (zhangjinpeng1987) [reply tests/test_rocksdb_options.rs]: Also need test when memtable flush to sst file, the content in wal writer's buffer will be dropped or not.

Comment (javaforfun) [reply tests/test_rocksdb_options.rs]: see:  https://github.com/facebook/rocksdb/blob/master/db/log_writer.h#L103 and https://github.com/facebook/rocksdb/blob/master/db/db_impl_write.cc#L1171
rocksdb alway flush the buffer of the last log before switching to a new one, so if memtable if full, WAL will be flushed

Comment (zhangjinpeng1987) [reply tests/test_rocksdb_options.rs]: I know it, but this test case test nothing? You should get the key after write to db.

Comment (javaforfun) [reply tests/test_rocksdb_options.rs]: I don't think we need do this in test_rocksdb_options.rs. 
eg. for test_sync_wal(), do we need simulate node crash, then read the key from db?

**Pull request merged**